scripts/pathseq/wdl/pathseq_pipeline.wdl

```diff
@@ -20,179 +26,712 @@
+
   # Mem is in units of GB but our command and memory runtime values are in MB
-  Int machine_mem = if defined(mem_gb) then mem_gb *1000 else default_ram_mb
-  Int command_mem = machine_mem - 4000
+  Int machine_mem = mem_gb * 1000
+  Int command_mem = ceil((machine_mem - size(filter_bwa_image, "MB")) * 0.8)
+
+  command <<<
+    set -euo pipefail
+    export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk4_jar_override}
+    touch ${filter_metrics_output_path}
+    gatk --java-options "-Xmx${command_mem}m" \
+      PathSeqFilterSpark \
+      --input ${input_bam_or_cram} \
+      --paired-output ${paired_bam_output_path} \
+      --unpaired-output ${unpaired_bam_output_path} \
+      --is-host-aligned ${is_host_aligned}  \
+      --verbosity ${verbosity} \
+      ${if gather_metrics then "--filter-metrics ${filter_metrics_output_path}" else ""} \
+      ${if defined(kmer_file) then "--kmer-file ${kmer_file}" else ""} \
+      ${if defined(filter_bwa_image) then "--filter-bwa-image ${filter_bwa_image}" else ""} \
+      ${if defined(bam_partition_size) then "--bam-partition-size ${bam_partition_size}" else ""} \
+      ${if defined(skip_quality_filters) then "--skip-quality-filters ${skip_quality_filters}" else ""}\
+      ${if defined(min_clipped_read_length) then "--min-clipped-read-length ${min_clipped_read_length}" else ""} \
+      ${if defined(max_masked_bases) then "--max-masked-bases ${max_masked_bases}" else ""} \
+      ${if defined(min_base_quality) then "--min-base-quality ${min_base_quality}" else ""} \
+      ${if defined(quality_threshold) then "--quality-threshold ${quality_threshold}" else ""} \
+      ${if defined(dust_mask_quality) then "--dust-mask-quality ${dust_mask_quality}" else ""} \
+      ${if defined(dust_window) then "--dust-window ${dust_window}" else ""} \
+      ${if defined(dust_t) then "--dust-t ${dust_t}" else ""} \
+      ${if defined(host_kmer_threshold) then "--host-kmer-thresh ${host_kmer_threshold}" else ""} \
+      ${if defined(max_adapter_mistmatches) then "--max-adapter-mismatches ${max_adapter_mistmatches}" else ""} \
+      ${if defined(min_adapter_length) then "--min-adapter-length ${min_adapter_length}" else ""} \
+      ${if defined(filter_bwa_seed_length) then "--filter-bwa-seed-length ${filter_bwa_seed_length}" else ""} \
+      ${if defined(host_min_identity) then "--host-min-identity ${host_min_identity}" else ""} \
+      ${if defined(filter_duplicates) then "--filter-duplicates ${filter_duplicates}" else ""} \
+      ${if defined(skip_pre_bwa_repartition) then "--skip-pre-bwa-repartition ${skip_pre_bwa_repartition}" else ""} \
+      ${if defined(filter_reads_per_partition) then "--filter-reads-per-partition ${filter_reads_per_partition}" else ""}
+
+    if [ ! -f "${paired_bam_output_path}" ]; then
+    	echo "File ${paired_bam_output_path} not found, creating empty BAM"
```

Comment:
Why do you prefer to create an empty BAM rather than have a default output path?

Reply:
If I remember correctly, the BAM will not be written if there are no reads. I believe this is a behavior of Spark tools in general and not something I can change in PathSeq.